MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -651,54 +647,24 @@ public void run() {
      * @param checkout a {@link Checkout} that has had a {@link CreditCard} associated with it using {@link #storeCreditCard(CreditCard, Checkout, Callback)}
      * @param callback the {@link Callback} that will be used to indicate the response from the asynchronous network operation, not null
      */
-    public void completeCheckout(Checkout checkout, final Callback<Checkout> callback) {
+    public void completeCheckout(Checkout checkout, final Callback<Payment> callback) {
         if (checkout == null) {
             throw new NullPointerException("checkout cannot be null");
         }
 
-        HashMap<String, String> requestBodyMap = new HashMap<>();
+        PaymentRequest paymentRequest = new PaymentRequest();
```

Comment:
we should just make a constructor that takes the sessionId, it is a write once object.